dev-tools/cloud/docker/build.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# This script builds an image from the elastic-agent image
+# with a locally built apm-server binary injected. Additional
+# flags (e.g. -t <name>) will be passed to `docker build`.
+
+set -eu
+
+REPO_ROOT=$(cd $(dirname $(readlink -f "$0"))/../../.. && pwd)
+
+USER_NAME=${USER}
+CI_ELASTIC_AGENT_DOCKER_IMAGE=docker.elastic.co/observability-ci/elastic-agent
+
+DEFAULT_IMAGE_TAG=8.8.0-681a23b0-SNAPSHOT
```

Comment:
Should this be left in place? Seems like we might need to update this over time to prevent accidentally deploying outdated versions by not specifying an argument. Maybe better to just not have a default here?

Reply:
it make things easier to use, passing and I think it's okay if the agent image is not always up to date as long the fleet server one, but we should probably update it sometimes. WDYT?

Reply:
Yeah let's just try not to let it fall too out of date that it's annoying or confusing to folks using this script.